base/array.jl

```diff
@@ -2318,7 +2318,7 @@ julia> filter(isodd, a)
  9
 ```
 """
-filter(f, As::AbstractArray) = As[map(f, As)::AbstractArray{Bool}]
+filter(f, As::AbstractArray) = @inbounds As[map(f, As)::AbstractArray{Bool}]
```

Comment:
This seems quite unlikely to make a difference, unless our inlining heuristic is messed up and we need to go fix that instead

Reply:
The `map` probably won’t be inlined, but I’m guessing the `getindex` may?

Obviously not a huge performance penalty but can help with tiny arrays I guess, so why not?

Reply:
Why might getindex get inlined here?

“Why not” isn’t really much of a reason to disable error checking for an unproven assertion for little expected gain

Reply:
OK I've removed this.

I still have *no* idea how `@inbounds As[map(f, As)::AbstractArray{Bool}]` could possibly be unsafe, unless someone completely borks up an implementation of `map`, but let's merge the other more useful change in this PR :)